rust/hyperlane-base/src/settings/mod.rs

```diff
@@ -251,12 +256,20 @@ impl Settings {
             name,
             self.metrics
                 .as_ref()
-                .map(|v| v.parse::<u16>().context("Port must be a valid u16"))
+                .map(|v| v.try_into().context("Port must be a valid u16"))
                 .transpose()?,
             prometheus::Registry::new(),
         )?))
     }
 
+    /// Make internal connections as-needed after deserializing.
+    pub(super) fn post_deserialize(&mut self) {
+        let Some(signer) = self.defaultsigner.take() else { return };
+        for chain in self.chains.values_mut() {
+            chain.signer.get_or_insert_with(|| signer.clone());
```

Comment:
Dumb rust question, if I specify a default signer *and* a signer in the chains object, which takes precedence?

Reply:
The default is only used if a signer for the chain is not specified. So the chain's signer takes precedence.